tool/planet/start.go

```diff
@@ -316,43 +323,54 @@ func addGroupToContainer(rootfs string, u serviceUser) error {
 	return trace.Wrap(err)
 }
 
-// setupCloudOptions sets up cloud flags and files passed to kubernetes
+// addCloudOptions sets up cloud flags and files passed to kubernetes
 // binaries, sets up container environment files
-func setupCloudOptions(c *Config) error {
+func addCloudOptions(c *Config) error {
 	if c.CloudProvider == "" {
 		return nil
 	}
 
-	flags := []string{fmt.Sprintf("--cloud-provider=%v", c.CloudProvider)}
+	contents, err := generateCloudConfig(c)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	c.Files = append(c.Files, box.File{
+		Path:     constants.CloudConfigFile,
+		Contents: strings.NewReader(contents),
+	})
+
+	c.Env.Upsert(EnvKubeletOptions, fmt.Sprintf("--cloud-provider=external"))
```

Comment:
I would add a comment with the excerpt from the documentation as to why this is needed. E.g. something like `In the cloud environment kubelet must run with --cloud-provider=external to ensure that it's aware that it must be initialized by the cloud controller manager before it is scheduled any work`.